### PR TITLE
Add a browser quickstart for the Firestore emulator

### PR DESCRIPTION
**Pull request by ryanpbrewster**
To get this to work you should:
  - start the Firestore emulator, with the flag `--webchannel-port=50051`
  - start the Firebase hosting emulator, via `firebase serve`
  - run the test via `npm test`

It brings up a headless instance of Chrome and manually interacts with the page elements, verifying that sending a message makes the full round trip through the Firestore emulator and appearing in the chat history.

### CODE REVIEW

File: firestore-emulator/browser-quickstart/firestore.indexes.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (samtstern) [inline firestore-emulator/browser-quickstart/firestore.indexes.json]: I think this file was included by mistake?  You could reduce it to:

```json
{
  "indexes": []
}
```

**Comment by samtstern** [comment]
@ryanpbrewster I tried to run a simple listen-with-limit against the emulator and got an error:

```
Jul 31, 2019 9:13:44 AM com.google.cloud.datastore.emulator.firestore.webchannel.FirestoreV1WebChannelAdapter$FirestoreListenHandler onMessage
WARNING: invalid message: {"database":"projects/fir-dumpster/databases/(default)","addTarget":{"query":{"structuredQuery":{"from":[{"collectionId":"items"}],"orderBy":[{"field":{"fieldPath":"__name__"},"direction":"ASCENDING"}],"limit":{"value":9}},"parent":"projects/fir-dumpster/databases/(default)/documents"},"targetId":2}}
Jul 31, 2019 9:13:44 AM com.google.net.webchannel.server.common.ChannelProcessor run
WARNING: Failed to consume the event.
java.lang.RuntimeException: com.google.protobuf.InvalidProtocolBufferException: Not an int32 value: {"value":9}
```

**Comment by ryanpbrewster** [comment]
wild, good catch, must be something to do with encoding integers into JSON

File: firestore-emulator/browser-quickstart/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Connecting a web app to the Firestore emulator
```
Comment (ryanpbrewster) [inline firestore-emulator/browser-quickstart/README.md]: @samtstern @markarndt thoughts on this README?

File: firestore-emulator/browser-quickstart/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Cloud Firestore relies heavily on bi-directional streams. Most clients connect
```
Comment (samtstern) [inline firestore-emulator/browser-quickstart/README.md]: For "most clients" say "most clients (such as Android, iOS, and Node.js)"

File: firestore-emulator/browser-quickstart/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+## Setup
+
+Download the Firestore emulator
+```
+curl -LO https://storage.googleapis.com/firebase-preview-drop/emulator/cloud-firestore-emulator-v1.7.0.jar
+```
+
+Start the Firestore emulator with WebChannel support enabled
+```
+java -jar cloud-firestore-emulator-v1.7.0.jar --webchannel-port=50051 --rules=firestore.rules
```
Comment (samtstern) [inline firestore-emulator/browser-quickstart/README.md]: What (if anything) is the default webchannel port?

Comment (ryanpbrewster) [reply firestore-emulator/browser-quickstart/README.md]: no default, if it's not provided then we don't accept webchannel traffic at all

**Pull request merged**